core/src/main/java/org/pentaho/di/core/row/ValueDataUtil.java

```diff
@@ -398,6 +415,56 @@ public static String createChecksum( ValueMetaInterface metaA, Object dataA, Str
     return md5Hash;
   }
 
+  public static String createChecksum( ValueMetaInterface metaA, Object dataA, String type, boolean failIfNoFile )
+          throws KettleFileException {
+    if ( dataA == null ) {
+      if ( failIfNoFile ) {
+        throw new KettleFileException();
+      }
+      return null;
+    }
+
+    String md5Hash = null;
+    FileObject file = null;
+    try {
+      file = KettleVFS.getFileObject( dataA.toString() );
+      throwsErrorOnFileNonExistence( file );
+      InputStream in = KettleVFS.getInputStream( file );
+      int bytes = in.available();
+      byte[] buffer = new byte[bytes];
+      in.read( buffer );
+
+      StringBuffer md5HashBuff = new StringBuffer( 32 );
+      byte[] b = MessageDigest.getInstance( type ).digest( buffer );
+      int len = b.length;
+      for ( int x = 0; x < len; x++ ) {
+        md5HashBuff.append( String.format( "%02x", b[x] ) );
+      }
+
+      md5Hash = md5HashBuff.toString();
+
+    } catch ( KettleFileException e ) {
+      if ( failIfNoFile ) {
```

Comment:
Are we sure this only happens on a missing file? KettleFileException sounds more generic